models/Makefile

```diff
@@ -1,6 +1,96 @@
-MODEL ?= MODEL=https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.1-GGUF/resolve/main/mistral-7b-instruct-v0.1.Q4_K_M.gguf
-IMAGE ?= quay.io/ai-lab/llama:latest
+# Add entries below as follows:
+# 1. find the model name and download link.
+# 2. Create a make target passing that info the `download-selected-model` target
+# 3. Add a normalizer make target, this will prime a model by chaging its file name to `model.file` which is what gets mounted in our server images
 
-.PHONY: build
-build:
-	podman build --build-arg ${MODEL} -f Containerfile -t ${IMAGE} .
+SELECTED_MODEL_DOWNLOAD_NAME ?= $(MISTRAL_MODEL_NAME)
+SELECTED_MODEL_DOWNLOAD_URL ?= $(MISTRAL_MODEL_URL)
+
+### Factory targets
+
+.PHONY: download-selected-model
+download-selected-model:
+	curl -H "Cache-Control: no-cache" -s -S -L -f $(SELECTED_MODEL_DOWNLOAD_URL) -z $(SELECTED_MODEL_DOWNLOAD_NAME) -o $(SELECTED_MODEL_DOWNLOAD_NAME).tmp && \
+	mv -f $(SELECTED_MODEL_DOWNLOAD_NAME).tmp $(SELECTED_MODEL_DOWNLOAD_NAME) 2>/dev/null || \
+	rm -f $(SELECTED_MODEL_DOWNLOAD_NAME).tmp $(SELECTED_MODEL_DOWNLOAD_NAME)
+
+.PHONY: normalize-model-file
+normalize-model-file: 
+	cp $(SELECTED_MODEL_DOWNLOAD_NAME) ./model.file
```

Comment:
This seems like it will obfuscate what model is being used and will overwrite a second model if its downloaded. Why do we need this normalization?

Reply:
The idea here was that way we can mount every model the same way which I thought would make our lives a bit easier, but I suppose in retrospect, the tiny amount of effort saved there is drastically outweighed by trying to debug a `model_server` without knowing what model its serving. Il revert this part.

Reply:
Also should this be a hard link? Followed by a cp if the hard link fails.

Reply:
@rhatdan I liked your implementation of a common directory so I redid the PR with that idea